src/cla/tcp.rs

```diff
@@ -0,0 +1,511 @@
+                    }
+                    let mut transfer_finished = false;
+                    loop {
+                        if let Some(packet) = self.receiver_task.recv().await {
+                            match packet {
+                                TcpClPacket::SessTerm(mut data) => {
+                                    connection_closing = true;
+                                    data.flags |= SessTermFlags::REPLY;
+                                    self.sender_task.send(TcpClPacket::SessTerm(data)).await?;
+                                }
+                                TcpClPacket::XferSeg(data) => {
+                                    vec.append(&mut data.buf.to_vec());
+                                    len += data.len;
+                                    transfer_finished = data.flags.contains(XferSegmentFlags::END);
+                                    self.sender_task
+                                        .send(TcpClPacket::XferAck(XferAckData {
+                                            tid: data.tid,
+                                            len,
+                                            flags: XferSegmentFlags::empty(),
+                                        }))
+                                        .await?;
+                                }
+                                TcpClPacket::MsgReject(data) => {
+                                    warn!("Received message reject: {:?}", data);
+                                }
+                                _ => {
+                                    error!("Unexpected packet");
+                                    connection_closing = true;
+                                }
+                            }
+                        } else {
+                            connection_closing = true;
+                        }
+                        if transfer_finished {
+                            break;
+                        }
+                    }
+                }
+                match Bundle::try_from(vec) {
+                    Ok(bundle) => {
+                        tokio::spawn(async move {
+                            if let Err(err) = crate::core::processing::receive(bundle).await {
+                                error!("Failed to process bundle: {}", err);
+                            }
+                        });
+                    }
+                    Err(err) => {
+                        error!("Failed to parse bundle: {}", err);
+                        self.sender_task
+                            .send(TcpClPacket::XferRefuse(XferRefuseData {
+                                reason: XferRefuseReasonCode::NotAcceptable,
+                                tid: data.tid,
+                            }))
+                            .await?;
+                    }
+                }
+                return Ok(connection_closing);
+            }
+            _ => {
+                error!("Unexpected packet");
```

Comment:
add a debug print giving a few more information, packet bytes, magic numbers, first 10 bytes or something to debug the issue.. which packet type was it?